tensorflow/core/kernels/data/padded_batch_dataset_op.cc

```diff
@@ -25,420 +28,403 @@ limitations under the License.
-      OP_REQUIRES(
-          ctx, TensorShapeUtils::IsScalar(padding_value_t.shape()),
-          errors::InvalidArgument("All padding values must be scalars"));
-      OP_REQUIRES(ctx, padding_value_t.dtype() == input->output_dtypes()[i],
-                  errors::InvalidArgument(
-                      "Mismatched type between padding value ", i,
-                      " and input dataset's component ", i, ": ",
-                      DataTypeString(padding_value_t.dtype()), " vs. ",
-                      DataTypeString(input->output_dtypes()[i])));
-      padding_values.push_back(tensor::DeepCopy(padding_value_t));
-    }
+  std::unique_ptr<IteratorBase> MakeIteratorInternal(
+      const string& prefix) const override {
+    return absl::make_unique<Iterator>(Iterator::Params{
+        this, name_utils::IteratorPrefix(kDatasetType, prefix)});
+  }
 
-    *output =
-        new Dataset(ctx, batch_size, drop_remainder, parallel_copy_,
-                    std::move(padded_shapes), std::move(padding_values), input);
+  const DataTypeVector& output_dtypes() const override {
+    return input_->output_dtypes();
   }
 
- private:
-  class Dataset : public DatasetBase {
-   public:
-    Dataset(OpKernelContext* ctx, int64 batch_size, bool drop_remainder,
-            bool parallel_copy, std::vector<PartialTensorShape> padded_shapes,
-            std::vector<Tensor> padding_values, const DatasetBase* input)
-        : DatasetBase(DatasetContext(ctx)),
-          batch_size_(batch_size),
-          drop_remainder_(drop_remainder),
-          parallel_copy_(parallel_copy),
-          padded_shapes_(std::move(padded_shapes)),
-          padding_values_(std::move(padding_values)),
-          input_(input) {
-      input_->Ref();
-
-      // NOTE(mrry): Currently we implement "batch up to" semantics. If we could
-      // tell statically that the input dataset is infinite, then we could
-      // always report `batch_size` as the 0th dimension.
-      //
-      // TODO(mrry): Need to validate that the input shape and the padded shape
-      // are "compatible" (i.e. that padded shape is >= input shape, with both
-      // static and dynamic checks as appropriate).
-      const auto& input_shapes = input_->output_shapes();
-      output_shapes_.reserve(input_shapes.size());
-      for (size_t i = 0; i < input_shapes.size(); ++i) {
-        if (drop_remainder_) {
-          output_shapes_.push_back(
-              PartialTensorShape({batch_size_}).Concatenate(padded_shapes_[i]));
-        } else {
-          output_shapes_.push_back(
-              PartialTensorShape({-1}).Concatenate(padded_shapes_[i]));
-        }
-      }
-    }
+  const std::vector<PartialTensorShape>& output_shapes() const override {
+    return output_shapes_;
+  }
 
-    ~Dataset() override { input_->Unref(); }
+  string DebugString() const override {
+    return name_utils::DatasetDebugString(kDatasetType, batch_size_);
```

Comment:
After refactoring, the output of `DebugString` will be a little different from before: `PaddedBatchDatasetV2Op(BATCH_SIZE)::Dataset` vs. `PaddedBatchDatasetOp(BATCH_SIZE)::Dataset`.

It is caused by that `name_utils::OpName(PaddedBatchDatasetOp::kDataType)` returns `PaddedBatchDatasetV2` instead of `PaddedBatchDataset`. 

The reason why`name_utils::OpName(PaddedBatchDatasetOp::kDataType)` return `PaddedBatchDatasetV2` is that the test file [here](https://github.com/tensorflow/tensorflow/pull/29098/files#diff-ae0aec972881c1a23a834cd178c0fb49R101) uses it as the op name to create `PaddedBatchDataset` op kernel. 

I'm wondering if `name_utils::OpName(PaddedBatchDatasetOp::kDataType)` should return `PaddedBatchDatasetV2` or `PaddedBatchDataset`?

Reply:
It should return "PaddedBatchDataset".